src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,156 @@
+                throw new ArgumentException("Address is null or empty.");
+
+            switch (transportType)
+            {
+                case TransportType.NamedPipe:
+                {
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{NamedPipeSchema} only supported on Windows.");
+                    break;
+                }
+                case TransportType.UnixDomainSocket:
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{UnixDomainSocketSchema} not supported on Windows, use {NamedPipeSchema}.");
+                    break;
+                }
+                default:
+                {
+                    throw new NotSupportedException($"{transportType} not supported.");
+                }
+            }
+
+            Address = address;
+            _transportType = transportType;
+            _portType = portType;
+        }
+
+        public static bool TryParse(string config, out IpcEndpointConfig result)
+        {
+            try
+            {
+                result = Parse(config);
+            }
+            catch(Exception)
+            {
+                result = null;
+            }
+            return result != null;
+        }
+
+        public static IpcEndpointConfig Parse(string config)
+        {
+            if (string.IsNullOrEmpty(config))
+                throw new FormatException("Missing IPC endpoint config.");
+
+            string address = "";
+            PortType portType = PortType.Connect;
+            TransportType transportType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? TransportType.NamedPipe : TransportType.UnixDomainSocket;
+
+            if (!string.IsNullOrEmpty(config))
+            {
+                var parts = config.Split(',');
+                if (parts.Length > 2)
+                    throw new FormatException($"Unknow IPC endpoint config format, {config}.");
+
+                if (string.IsNullOrEmpty(parts[0]))
+                    throw new FormatException($"Missing IPC endpoint config address, {config}.");
+
+                portType = PortType.Listen;
+                address = parts[0];
+
+                if (parts.Length == 2)
+                {
+                    if (string.Equals(parts[1], "connect", StringComparison.OrdinalIgnoreCase))
+                    {
+                        portType = PortType.Connect;
+                    }
+                    else if (string.Equals(parts[1], "listen", StringComparison.OrdinalIgnoreCase))
+                    {
+                        portType = PortType.Listen;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Unknow IPC endpoint config keyword, {parts[1]} in {config}.");
+                    }
```

Comment:
```suggestion
                portType = parts[1].ToLower() switch
                {
                	"connect" => PortType.Connect,
                	"listen" => PortType.Listen,
                	_ => throw new FormatException($"Unknown IPC endpoint config keyword, {parts[1]} in {config}.");
                };
```

I'm a fan of the switch expression for this kind of stuff due to compactness, but it certainly isn't necessary to switch.

Reply:
If changing to switch, check that parts[1] is not null before using ToLower().

Reply:
I will keep the if/else if/else for now.